src/recommendpolicy/helperFunctions.go

```diff
@@ -189,3 +207,65 @@ func initDeploymentWatcher() {
 		}
 	}
 }
+
+func mergePolicy(ms types.MatchSpec, name, namespace string, labels LabelMap) (types.KnoxSystemPolicy, error) {
+	policy := types.KnoxSystemPolicy{
+		Spec: types.KnoxSystemSpec{
+			Selector: types.Selector{
+				MatchLabels: map[string]string{}},
+			Process: types.KnoxSys{
+				MatchDirectories: []types.KnoxMatchDirectories{
+					{Dir: "",
+						Action:   ms.Spec.Action,
+						Severity: ms.Spec.Severity,
+						Message:  ms.Spec.Message,
+						Tags:     ms.Spec.Tags,
+					},
+				},
+				MatchPaths: []types.KnoxMatchPaths{
+					{
+						Action:   ms.Spec.Action,
+						Severity: ms.Spec.Severity,
+						Message:  ms.Spec.Message,
+						Tags:     ms.Spec.Tags,
+					},
+				},
+			},
+			File: types.KnoxSys{
+				MatchDirectories: []types.KnoxMatchDirectories{
+					{
+						Action:   ms.Spec.Action,
+						Severity: ms.Spec.Severity,
+						Message:  ms.Spec.Message,
+						Tags:     ms.Spec.Tags,
+					},
+				},
+				MatchPaths: []types.KnoxMatchPaths{
+					{
+						Action:   ms.Spec.Action,
+						Severity: ms.Spec.Severity,
+						Message:  ms.Spec.Message,
+						Tags:     ms.Spec.Tags,
+					},
+				},
+			},
+		},
+	}
+	policy.APIVersion = "v1"
+	policy.Kind = ms.Kind
+
+	if policy.Kind != types.KindKubeArmorHostPolicy {
+		policy.Metadata = map[string]string{
+			"name":      fmt.Sprintf("%v-%v-%v", types.HardeningMergePolicy, name, ms.Name),
+			"namespace": namespace,
+		}
+	} else {
+		policy.Metadata = map[string]string{
+			"name": fmt.Sprintf("%v-host-%v-%v", types.HardeningMergePolicy, name, ms.Name),
+		}
+	}
+	policy.Spec.Selector.MatchLabels = labels
+
+	addPolicyRule(&policy, &ms.Spec)
```

Comment:
passing this data to addPolicyRule is not necessary right? since both functions are doing almost similar job

Reply:
that has been rearranged.